Library/Homebrew/formulary.rb

```diff
@@ -638,17 +736,83 @@ def load_file(flags:, ignore_errors:)
       e.issues_url = tap.issues_url || tap.to_s
       raise
     end
+  end
 
-    private
+  # Loads a formula from a name, as long as it exists only in a single tap.
+  module FromNameLoader
+    sig {
+      params(ref: T.any(String, Pathname, URI::Generic), from: Symbol, warn: T::Boolean)
+        .returns(T.nilable(FromTapLoader))
+    }
+    def self.try_new(ref, from: T.unsafe(nil), warn: false)
+      return unless ref.is_a?(String)
+      return if ref.include?("/")
+
+      name = ref
+
+      loaders = Tap.map { |tap| FromTapLoader.try_new("#{tap}/#{name}") }
+                   .compact
+                   .select { _1.path.exist? }
+
+      case loaders.count
+      when 1
+        loaders.first
+      when 2..Float::INFINITY
+        # Always prefer the default tap, i.e. behave the same as if loading from the API.
+        if (default_tap_loader = loaders.find { _1.tap.core_tap? })
+          return default_tap_loader
+        end
 
-    def find_formula_from_name(name, tap)
-      Formulary.find_formula_in_tap(name, tap)
+        raise TapFormulaAmbiguityError.new(name, loaders.map(&:tap))
```

Comment:
```suggestion
        if type == :rename
          raise TapFormulaWithOldnameAmbiguityError.new(???)
        else
          raise TapFormulaAmbiguityError.new(name, loaders.map(&:tap))
        end
```

Or alternatively burn `TapFormulaWithOldnameAmbiguityError` entirely, though this PR is already quite large.

Reply:
Looking at the old logic, it seems like this expects the same old name to occur in multiple taps, rather than a single rename in the default tap conflicting with other taps' formula names. This seems weirdly specific, so I think removing `TapFormulaWithOldnameAmbiguityError` makes more sense.